### PR TITLE
Update Gradle to 5.5 version

### PR DESCRIPTION
**Pull request by chetanmeh**
Updates Gradle to 5.5 version and also updates Gradle Scoverage plugin to 3.1.5

## Description

For enabling running and building on JDK 11 (#4217) we need to move to Gradle 5.x version which officially supports JDK 11. In doing so we would need to update the [Gradle Scoverage](https://github.com/scoverage/gradle-scoverage) plugin to 3.1.5 as existing version was not compatible

### Benefits

1. Enables building on JDK 11
2. Gradle 5.x has better support for Maven publishing. We would now be needing that to publish the standalone jar

### Scoverage Version Impact

Newer soverage plugin has a different model of working and does not produce a jar of instrumented classes which impacts our logic for coverage collection for code running within container (#3685). To make it work we now copy the instrumented classes within container and remove existing packaged jars from the lib

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#4217 )

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [ ] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [ ] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: tests/build.gradle [hunk 1]
```diff
@@ -250,8 +255,33 @@ gradle.projectsEvaluated {
...
             exceptionFormat = 'full'
         }
+        maxHeapSize = "1500m"
         outputs.upToDateWhen { false } // force tests to run every time
     }
+    /**
+     * Task to generate coverage xml report. Requires the
+     * tests to be executed prior to its invocation
+     */
+    task reportCoverage(type: ScoverageReport) {
```
Comment (chetanmeh) [inline tests/build.gradle]: The new version now creates task like `report<Test Task Name>Scoverage` as entry task. This task in turn adapts the compile task to use the instrumented classes. This works fine for conventional cases where tests are part of same module. 

However in our case as we have a separate test module and further to capture coverage of code running within container (as part of integration test) we copy the coverage metrics files manually we cannot use default tasks. Hence the report task duplicate bit of logic from actual task

File: tests/build.gradle [hunk 2]
```diff
@@ -250,8 +255,33 @@ gradle.projectsEvaluated {
             showStandardStreams = true
             exceptionFormat = 'full'
         }
+        maxHeapSize = "1024m" //Gradle 5.5 defaults to 512MB which is low
```
Comment (chetanmeh) [inline tests/build.gradle]: With Gradle 5.0 it now [defaults to 512M of heap][1]. Due to this OOM was seen in test runs on Travis (which has 7 GB memory). Hence specifying the size explicitly to 1024 MB now

> All workers, including compilers and test executors, now start with 512MB of heap. The previous default was 1/4th of physical memory. Large projects may have to increase this setting on the relevant tasks, e.g. JavaCompile or Test.



[1]: https://docs.gradle.org/current/userguide/upgrading_version_4.html#rel5.0:default_memory_settings

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -17,6 +17,6 @@
 
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.5.1-all.zip
```
Comment (chetanmeh) [inline gradle/wrapper/gradle-wrapper.properties]: Switching to 5.5.1 and also to `all` mode so as to enable code assists in IDE

File: settings.gradle [hunk 4]
```diff
@@ -43,13 +43,6 @@ gradle.ext.scalafmt = [
     config: new File(rootProject.projectDir, '.scalafmt.conf')
 ]
 
-gradle.ext.scoverage = [
```
Comment (chetanmeh) [inline settings.gradle]: Now scoverage plugin internally adapts the classpath with required dependencies. So no need to explicitly add them

File: tests/build.gradle [hunk 5]
```diff
@@ -315,37 +330,11 @@ def loadWhiskProps(){
     p
 }
 
-/**
- * Aggregates the scoverage xml reports from various modules into a
- * single report
- */
-task aggregateCoverage(type: JavaExec, dependsOn: reportCoverage) {
```
Comment (chetanmeh) [inline tests/build.gradle]: This task was never used as we rely on [codecov][1] to perform actual aggregation

[1]: https://codecov.io/

**Comment by chetanmeh** [comment]
This PR is now ready for review

**Comment by chetanmeh** [comment]
@dgrove-oss @rabbah Can you review this PR. Once merged we can then work on publishing Maven artifacts from core repo (#4579)

File: gradlew.bat [hunk 6]
```diff
@@ -30,7 +30,7 @@ set APP_BASE_NAME=%~n0
 set APP_HOME=%DIRNAME%
 
 @rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-set DEFAULT_JVM_OPTS=-Dfile.encoding=UTF-8
+set DEFAULT_JVM_OPTS=-Dfile.encoding=UTF-8 "-Xmx64m" "-Xms64m"
```
Comment (rabbah) [inline gradlew.bat]: is this too low?

**Comment by rabbah** [comment]
No issues on my side using this patch.

File: gradlew.bat (hunk 6 above)
Comment (chetanmeh) [reply gradlew.bat]: Yeah but thats the [current default in there generated files][1]

[1]: https://github.com/gradle/gradle/blob/362226499afd9180ebc19822d44fca5fbaf004e9/gradlew#L47

Comment (dgrove-oss) [reply gradlew.bat]: If a different heap size is appropriate, we can change it.  It's fine for us to diverge from the upstream gradlew[.bat] as needed.

Comment (chetanmeh) [reply gradlew.bat]: For now build seems to work fine. If we see any issue then we can change. I believe Gradle wrapper internally forks a new jvm process so it needs bare minimum memory for its work

**Pull request merged**